tests/unit/plugins/test_enricher.py

```diff
@@ -19,18 +19,132 @@
 import pytest  # NOQA
 from gordon import interfaces
 
+from gordon_gcp import exceptions
 from gordon_gcp.plugins import enricher
+from gordon_gcp.plugins import event_consumer
 
 
-def test_implements_interface():
+INSTANCE_NETWORK_DATA = {
+    'networkInterfaces': [{
+        'accessConfigs': [{
+            'natIP': '35.229.173.136'
+        }],
+    }]
+}
+
+
+@pytest.fixture
+def config(fake_keyfile):
+    return {
+        'keyfile': fake_keyfile,
+        'scopes': [],
+        'dns_zone': 'example.com',
+        'retries': 5
+    }
+
+
+def test_implements_interface(config):
     """GCEEnricher implements IEnricherClient"""
-    config = {'foo': 'bar'}
     success, error = asyncio.Queue(), asyncio.Queue()
-    client = enricher.GCEEnricher(config, success, error)
+    client = enricher.GCEEnricher(config, None, success, error)
 
     assert interfaces.IEnricherClient.providedBy(client)
     assert interfaces.IEnricherClient.implementedBy(enricher.GCEEnricher)
     assert config is client.config
     assert success is client.success_channel
     assert error is client.error_channel
     assert 'enrich' == client.phase
+
+
+@pytest.fixture
+def event_msg(mocker):
+    fake_data = {
+            'action': 'additions',
+            'resourceName': 'projects/project1/zones/zone1/instances/instance1',
+            'resourceRecords': [],
+            'timestamp': '2017-12-04T20:13:44.181Z',
+        }
+    pubsub_msg = mocker.Mock()
+    pubsub_msg._ack_id = '1:1'
+    return event_consumer.GEventMessage(
+        pubsub_msg, fake_data, enricher.GCEEnricher.phase)
+
+
+@pytest.fixture
+def channel_pair():
+    return asyncio.Queue(), asyncio.Queue()
+
+
+@pytest.fixture
+def mock_http_client(mocker, get_mock_coro):
+    get_json_mock, get_json_coro = get_mock_coro()
+    http_client = mocker.Mock()
+    mocker.patch.object(http_client, 'get_json', get_json_coro)
+    mocker.patch.object(http_client, '_get_json', get_json_mock)
```

Comment:
Hmm this is a little confusing. Maybe instead of `_get_json`, `get_json.mock` or `get_json_mock` if not possible? I actually looked up if there was a `_get_json` method in the http client, lol.

Reply:
Thanks for pointing it out. Changed it to `get_json_mock`.

> I actually looked up if there was a _get_json method in the http client, lol.

GOT YOU! :grin: